File: Level-3 450 set  Pan-1/stack.py

# Push, Pop, Peek, IsEmpty, PrintStack
class Stack:
    def __init__(self):
        self.arr = []
        self.top = -1


    def push(self, x):
        self.arr.append(x)
        self.top += 1
    
    def pop(self):
        if self.top == -1:
            print("The Stack is empty!")
        else:
            print("Popped element is: ", self.arr[self.top])
            self.arr.pop(-1)
            self.top -= 1
    
    def peek(self):
        if self.top == -1:
            print("The Stack is empty!")
        print("Peeked: ", self.arr[self.top])

    def IsEmpty(self):
        if self.top == -1:
            print("The Stack is empty!")
        else:
            print("No")
    
    def printStack(self):
        if self.top <= -1:
            print("The Stack is empty!")
        for i in self.arr:
            print(i, end=" ")
        print("")











if __name__ == '__main__':
    stamp = Stack()
    stamp.push(1)
    stamp.push(2)
    stamp.push(3)
    stamp.push(4)
    stamp.printStack()
    stamp.pop()
    stamp.peek()
    stamp.IsEmpty()
    stamp.printStack()
    stamp.pop()
    stamp.pop()
    stamp.pop()
    stamp.pop()
    stamp.printStack()
    stamp.pop()

    

File: Book Coding/Placement series AC/stack.cpp
#include <iostream>
#include <bits/stdc++.h>
using namespace std;

#define n 5


class Stack{
    int* arr; 
    int top;

    public:
    Stack(){
        arr = new int[n];
        top = -1;
    }

    void push(int x){
        if (top == n-1){
            cout << "Stack Overflow"<<endl;
            return;
        }

        top++;
        arr[top] = x;

    }

    void pop(){
        if (top < 0){
            cout << "Stack Overflow"<<endl;
            return;
        }
        cout << "Popped element is : "<<arr[top]<<endl;
        top--;
    }

    void peek(){
        if (top<0){
            cout << "Its empty Madam/sir"<<endl;
            return;
        }
        cout << arr[top]<<endl;
    }

    void isEmpty(){
        if (top==0){
            cout << "Yes its empty"<<endl;
            return;
        }
        cout << "Its not empty"<<endl;
    }

    void printStack(){
        if (top==0){
            cout << "Its empty"<<endl;
            return;
        }
        for (int i=0; i<top; i++){
            cout << arr[i]<<" ";
        }
        cout<<endl;
    }

};



int main(){
    Stack stump;
    stump.push(1);
    stump.push(2);
    stump.push(6);
    stump.push(3);
    stump.push(3);
    stump.push(3);
    stump.push(3);


    stump.printStack();
    stump.isEmpty();
    stump.peek();
    stump.pop();
    stump.printStack();

    return 0;
}